heartbeat/monitors/wrappers/summarizer/summarizer.go

```diff
@@ -84,61 +120,48 @@ func NewJobSummary(attempt uint16, maxAttempts uint16, retryGroup string) *JobSu
 	}
 }
 
+// BumpAttempt swaps the JobSummary object's pointer for a new job summary
+// that is a clone of the current one but with the Attempt field incremented.
+func (js *JobSummary) BumpAttempt() {
+	*js = *NewJobSummary(js.Attempt+1, js.MaxAttempts, js.RetryGroup)
+}
+
 // Wrap wraps the given job in such a way that the last event summarizes all previous events
 // and additionally adds some common fields like monitor.check_group to all events.
 // This adds the state and summary top level fields.
 func (s *Summarizer) Wrap(j jobs.Job) jobs.Job {
 	return func(event *beat.Event) ([]jobs.Job, error) {
-		conts, jobErr := j(event)
-
-		_, _ = event.PutValue("monitor.check_group", fmt.Sprintf("%s-%d", s.checkGroup, s.jobSummary.Attempt))
+		conts, eventErr := j(event)
 
 		s.mtx.Lock()
 		defer s.mtx.Unlock()
 
-		js := s.jobSummary
-
 		s.contsRemaining-- // we just ran one cont, discount it
 		// these many still need to be processed
 		s.contsRemaining += uint16(len(conts))
 
-		monitorStatus, err := event.GetValue("monitor.status")
-		if err == nil && !eventext.IsEventCancelled(event) { // if this event contains a status...
-			mss := monitorstate.StateStatus(monitorStatus.(string))
-
-			if mss == monitorstate.StatusUp {
-				js.Up++
-			} else {
-				js.Down++
+		for _, plugin := range s.plugins {
+			actions := plugin.EachEvent(event, eventErr)
+			if actions&DropErrEvent != 0 {
```

Comment:
If we are treating the EachEvent similar to exit codes, can we document that. Its easier to follow though.

Reply:
I'm not sure I follow. I was a bit split on returning a bit field instead of just a single code. It's a moot point since there's only one thing, and I don't think one is more right than the other.

Reply:
I was asking if we are treating actions as 0 as success and 1 as Error.